product_route_profile/models/product_template.py

```diff
@@ -0,0 +1,35 @@
+# Copyright 2022 Akretion (https://www.akretion.com).
+# @author Kévin Roche <kevin.roche@akretion.com>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    route_profile_id = fields.Many2one("route.profile", string="Route Profile")
+    force_route_profile_id = fields.Many2one(
+        "route.profile",
+        string="Priority Route Profile",
+        company_dependent=True,
+        help="If defined, the "
+        "priority route profile will be used and will replace the "
+        "route profile, only for this company.",
+    )
+
+    route_ids = fields.Many2many(
+        compute="_compute_route_ids",
+        store=False,
+    )
+
+    @api.depends("route_profile_id", "force_route_profile_id")
+    @api.depends_context("company")
```

Comment:
Where does it depend on this ?

Reply:
`force_route_profile_id` is company dependent

Reply:
Yes, saw it. But doesn't Odoo ORM aware of that through that field ?

Your new depends IMHO should not add that entry.

Reply:
After some checks, it appears to be unnecessary, I remove it.

Reply:
After deeper investigation, @api.depends_context("company") is relevant here, as we can see with the fail test (previous commit) and the fix in the last commit.

Reply:
Ok, I think this is a problem in ORM as company_dependant fields that are in depends should imply that behaviour. But ok